### PR TITLE
fix(image-tags): use unique timestamp to build image

### PR DESCRIPTION
**Pull request by MartinBasti**
Unique timestamp is the only way to avoid rewrting daily images.

Images from PRs will have special "PR-" prefix in tag

### CODE REVIEW

**Review by jsztuka — APPROVED** [review]
lgtm

**Comment by MartinBasti** [comment]
it's not issue in buildah, something is wrong in PR

File: .tekton/build_and_trigger.yaml [hunk 1]
```diff
@@ -49,19 +49,51 @@ spec:
...
+                  echo "PR-$(params.revision)" | tee $(results.image_tag.path)
+                fi
       - name: build-image
         workspaces:
           - name: source
             workspace: workspace
         params:
           - name: IMAGE
-            value: quay.io/redhat-appstudio/clair-in-ci:{{revision}}
+            value: "quay.io/redhat-appstudio/clair-in-ci:$(tasks.calculate-tag.results.image_tag)"
```
Comment (MartinBasti) [inline .tekton/build_and_trigger.yaml:90]: This line causes issues, I don't knwo why.

When I add `{{revision}}` back it works.
Also I don't understand what exactly is wrong, first half of buildah task just works fine and uses value from `image_tag`

Comment (MartinBasti) [reply .tekton/build_and_trigger.yaml:90]: This is the error, build passed, but task didn't get to push part. You  can also see, that image_tag value has been used there, so I really don't know why it failed later.
```
COMMIT quay.io/redhat-appstudio/clair-in-ci:PR-224ee76d32e10624fb88417d1973c967be682076
--> f92d40dd1f1
Successfully tagged quay.io/redhat-appstudio/clair-in-ci:PR-224ee76d32e10624fb88417d1973c967be682076
f92d40dd1f13f94d375065b6f99b06fce40c5c3c58bec9e8f065b228f50b659b
/tekton/scripts/script-0-rvhs2: line 7: .: filename argument required
.: usage: . filename [arguments]
```

File: .tekton/build_and_trigger.yaml [hunk 2]
```diff
@@ -49,19 +49,51 @@ spec:
...
+          steps:
+            - name: make-tag
+              image: registry.access.redhat.com/ubi9
+              script: |
+                #!/usr/bin/env bash
+                VERSION_MAJOR=v1
+                if [ "$(params.target_branch)" == "daily" ] || [ "$(params.event_type)" == "push" ]; then
+                  echo "${VERSION_MAJOR}.$(date --utc '+%Y%m%d%H%M%S')" | tee $(results.image_tag.path)
+                else
+                  echo "PR-$(params.revision)" | tee $(results.image_tag.path)
```
Comment (dirgim) [inline .tekton/build_and_trigger.yaml:82]: Maybe using `echo -n` would resolve the issue in buildah task?

**Review by dirgim — APPROVED** [review]
LGTM

**Pull request merged**